verilog/analysis/flow_tree.h

```diff
@@ -0,0 +1,114 @@
+// Copyright 2017-2022 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_FLOW_TREE_H_
+#define VERIBLE_VERILOG_FLOW_TREE_H_
+
+#include <bitset>
+#include <map>
+#include <stack>
+#include <string>
+#include <vector>
+
+#include "absl/status/status.h"
+#include "common/lexer/token_stream_adapter.h"
+#include "verilog/parser/verilog_token_enum.h"
+
+namespace verilog {
+using TokenSequenceConstIterator = verible::TokenSequence::const_iterator;
+
+// Receive a complete token sequence of one variant.
+// 1st argument: generated variant.
+// 2nd argument: number of generated variants (to decide if more is needed).
```

Comment:
I'd remove this parameter. You already can deal with 2^128 different variants, but you only provide a number that can hold 2^31 positive values.

I'd leave it out for now. If we ever want to tell the callback which exact variant we're talking about, we might need to pass the bitset (as reference). But I guess for now let's not worry about it.